crates/solidity/outputs/cargo/tests/src/cst_output/generated/mod.rs

```diff
@@ -25,26 +23,16 @@ mod Expression;
 #[allow(non_snake_case)]
 mod FallbackFunctionDefinition;
 #[allow(non_snake_case)]
-mod FixedBytesType;
-#[allow(non_snake_case)]
 mod FunctionDefinition;
```

Comment:
if these are removed from parser tests, where are the new entries for scanner tests?

Reply:
I just tried modifying a snapshot, rerunning tests, which passed without any changes.

Reply:
Scanner tests removed

Reply:
Not sure I understand the reason behind that. Is it because of the new lexical contexts? can we split the tests per scope to handle that?

- `cst_output/parser/PARSER_NAME/TEST_NAME`
- `cst_output/scanner/LEXICAL_SCOPE_NAME/SCANNER_NAME/TEST_NAME`

These are tens of tests that cover a good number of corner cases we discovered over the last year. With our coverage already not where we want it to be, I'm afraid that would allow more breaks to happen with any grammar migrations. We already discovered a few in this PR.

Reply:
It's because scanner functions no longer exist. Scanners are now next_token and are better handled IMO by tests like I have in the explicit scanner tests in e.g. crates/solidity/outputs/cargo/tests/src/scanner/mod.rs

Reply:
I don't think the new `crates/solidity/outputs/cargo/tests/src/scanner/mod.rs` covers all the test cases that were deleted. Additionally, I think we are fine with one manual test like this, but it is much harder to inspect/update/debug, compared to our snapshot infra, so I won't suggest adding more.

I wonder what blocks using `*_next_token` in the snapshot tests? Can't we parse the next token, then assert it is the intended type?

Reply:
F2f: let's keep the old scanner tests, even if they are unused now, and I will migrate them to run under the closest parser in a following PR.

Reply:
done